src/defrag.c

```diff
@@ -1323,10 +1322,39 @@ static doneStatus defragStagePubsubKvstore(void *ctx, monotime endtime) {
         defragPubsubScanCallback, NULL, &defragfns);
 }
 
+/* Defragmentation callback for Lua scripts dictionary. */
+void activeDefragScriptsDictCallback(void *privdata, const dictEntry *de, dictEntryLink plink) {
+    UNUSED(plink);
+    dict *d = privdata;
+    sds newkey, key = dictGetKey(de);  /* Get the script SHA key (sds string) */
+    luaScript *l = dictGetVal(de);   /* Get the luaScript structure */
+
+    if ((newkey = activeDefragSds(key))) {
+        dictSetKey(d, (dictEntry *)de, newkey);
+
+        /* Update the LRU list node value to point to the new key.
+         * The LRU list stores SHA strings as node values, so we need to
+         * update the reference when the key is reallocated during defragmentation. */
+        if (l->node)
+            l->node->value = newkey;
```

Comment:
was this update missing (regardless of defrag being called within a blocked script)?

Reply:
btw, we're not defragging the list nodes, right?
maybe we should.
and maybe we should just avoid defrag from within blocked scripts (don't recall if there was a good reason we did it)

Reply:
ohh, the reason was AOF loading (we must to defrag during AOF loading).
if we want, we can certainly isolate it so it only happens then, and not during script and RDB.

Reply:
i agree we should separate between the (critical) bug fix, and the other improvements.
in that light, maybe we can change the fix in a way that it will be more backport compatible?